WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/CouponInputTransformer.swift

```diff
@@ -5,33 +5,24 @@ import Yosemite
 ///
 struct CouponInputTransformer {
 
-    /// Adds, deletes, or updates a coupon line input into an existing order.
-    /// Only the first coupon line is updated.
+    /// Adds a coupon line input into an existing order.
+    /// If the order already has that coupon line it does nothing
     ///
-    /// - If input is `nil`, then we remove the first existing coupon line.
-    /// - If there is no existing coupon lines, we insert the input one.
-    /// - If there are existing coupon lines replace the first coupon line
-    ///
-    static func update(input: OrderCouponLine?, on order: Order) -> Order {
-        // If input is `nil`, then we remove the first existing coupon line.
-        guard let input = input else {
-            let updatedLines = order.coupons.enumerated().map { index, line -> OrderCouponLine? in
-                if index == 0 {
-                    return nil
-                }
-                return OrderFactory.newOrderCouponLine(code: line.code)
-            }.compactMap({ $0 })
-            return order.copy(coupons: updatedLines)
+    static func append(input: String, on order: Order) -> Order {
+        guard !order.coupons.map({ $0.code }).contains(input) else {
+            return order
         }
 
-        // If there is no existing coupon lines, we insert the input one.
-        guard order.coupons.isNotEmpty else {
-            return order.copy(coupons: [input])
-        }
+        return order.copy(coupons: order.coupons + [OrderFactory.newOrderCouponLine(code: input)])
+    }
+
+    /// Removes a coupon line input into an existing order.
```

Comment:
```suggestion
    /// Removes a coupon line input from an existing order.
```

Reply:
Done in b63aeabf4ffabf9e37c63cd69e445154c1a718f9